geolocation-API/getCurrentPosition_IDL.https.html

```diff
@@ -1,132 +1,111 @@
-<!DOCTYPE HTML>
-<meta charset="utf-8">
-<meta name=timeout content=long>
+<!DOCTYPE html>
+<meta charset="utf-8" />
+<meta name="timeout" content="long" />
 <title>Geolocation Test: getCurrentPosition tests</title>
-<link rel="help" href="http://www.w3.org/TR/geolocation-API/">
+<link rel="help" href="http://www.w3.org/TR/geolocation-API/" />
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
-
-<p>Clear all Geolocation permissions before running this test. If prompted for permission, please allow.</p>
-<div id="log"></div>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
 <script>
-var geo, success, fail;
-
-setup(function() {
-  geo = navigator.geolocation;
-}, {explicit_done: true});
-
-function successCallback(position)
-{
-  var ii, oldval;
-
-  /*
+  const resetPermission = () => {
+    return test_driver.set_permission({ name: "geolocation" }, "prompt");
+  };
+  promise_test(async (t) => {
+    t.add_cleanup(resetPermission);
+    await test_driver.set_permission({ name: "geolocation" }, "denied");
+    const promise = new Promise((resolve, reject) => {
+      navigator.geolocation.getCurrentPosition(reject, resolve);
+    });
+    /*
+    interface GeolocationPositionError {
+      const unsigned short PERMISSION_DENIED = 1;
+      const unsigned short POSITION_UNAVAILABLE = 2;
+      const unsigned short TIMEOUT = 3;
+      readonly attribute unsigned short code;
+      readonly attribute DOMString message;
+    };
+    */
+    const error = await promise;
+    assert_true(
+      error instanceof GeolocationPositionError,
+      "expected GeolocationPositionError"
+    );
+    assert_equals(GeolocationPositionError.PERMISSION_DENIED, 1);
+    assert_equals(GeolocationPositionError.POSITION_UNAVAILABLE, 2);
+    assert_equals(GeolocationPositionError.TIMEOUT, 3);
+    assert_equals(error.code, GeolocationPositionError.PERMISSION_DENIED);
```

Comment:
This is also tested in `getCurrentPosition_permission_deny.https.html` so I'd just check its type instead.